src/Lecture3.hs

```diff
@@ -82,7 +82,10 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+next :: (Ord a, Enum a, Bounded a) => a -> a
+next x
+    | (maxBound `asTypeOf` x) == x = (minBound `asTypeOf` x)
```

Comment:
You don't need to use `asTypeOf` here because in this case Haskell figures out proper types due to type inference 👍🏻 

```suggestion
    | maxBound == x = minBound
```